update Findings page (#1384)
semgrep/semgrep-docs: docs/semgrep-code/findings.md

}
</ul>

Semgrep Code generates a **finding** when a rule matches a piece of code in your codebase. You can use Semgrep Cloud Platform's [**Findings** page](https://semgrep.dev/orgs/-/findings) page to view all of the findings generated by Semgrep Code after it scans your codebase.

![Semgrep Cloud Platform Findings page](/img/app-findings-overview.png#md-width)<br />

## Navigate to the Findings page

To access the [**Findings** page](https://semgrep.dev/orgs/-/findings):

1. [Log into Semgrep Cloud Platform](https://semgrep.dev/login)
2. Click **[Code](https://semgrep.dev/orgs/-/findings)** to navigate to the **Findings** page.

## Findings page structure

The **Findings** page consists of:

> - The filter panel, which you can use to group and filter for specific findings

Reviewer: ```suggestion
- The **filter panel**, which you can use to group and filter for specific findings
```

- Information about findings identified by Semgrep Code. Each  finding in the list includes:
  - The name and description of the rule used to generate the finding
  - The name of the project, as well as links to the Git branch and source code, where Semgrep Code identified the finding

### Group findings

> By default, Semgrep groups all of the findings by the rule Semgrep used to match the code. The rules that reported the highest number of findings are at the top of the page. This helps you see which rules have the highest efficiency and which generate a lot of noise in your results.

Reviewer: ```suggestion
By default, Semgrep groups all of the findings by the rule Semgrep used to match the code. This view is called the **Group by rule** view. The rules that reported the highest number of findings are at the top of the page. This helps you see which rules have the highest efficiency and which generate a lot of noise in your results.
```

> ![Screenshot of the Findings page with findings grouped by rule](/img/app-findings.png#md-width)

Reviewer: ```suggestion
![Findings grouped by rule](/img/app-findings.png#md-width)
```

To view findings individually, toggle **Group by Rule** to **No grouping** using the drop-down menu in the header. Findings are displayed based on the date they were found, with the most recent finding listed at the top.

> ![Screenshot of the Group by Rule option](/img/cloud-platform-findings-no-grouping.png#md-width)

Reviewer: ```suggestion
![Group by Rule option](/img/cloud-platform-findings-no-grouping.png#md-width)
```

### Filter findings

> Semgrep offers multiple filters you can use to narrow down your results. The following criteria are available for filtering:

Reviewer: ```suggestion
Use filters to narrow down your results. The following criteria are available for filtering:
```

| Filter                 | Description  |
| ---------------------  | ------------ |

> | **Projects**           | Filter by codebases connected to Semgrep Cloud Platform. |

Reviewer: ```suggestion
| **Projects**           | Filter by repositories connected to Semgrep Cloud Platform. |
```
Anything that can be scanned and uploaded to Semgrep Cloud Platform (`semgrep ci`) is a git repo.

        </dd>
    </dl>

:::tip
You can identify findings categorized under **Security** by their badge.
![Screenshot of security badge](/img/findings-security-badge.png#md-width)
:::

### Display findings reported in a specific timeframe

By default, the **Findings** page displays your results from all time. To display findings reported during a specific timeframe, click the <i class="fa-solid fa-calendar-days"></i> **All time** button and select the preferred period. The following options are available:

  - All time
  - Last 1 year
  - Last 1 month
  - Last 7 days
  - Last 1 day

### View findings details about a specific finding

To view in-depth information about a specific finding:

1. Select the finding whose details you want to view details:

>     - If the default **Group by Rule** is enabled, click <i class="fa-regular fa-window-restore"></i> **Details** icon on the card of the finding.

Reviewer: ```suggestion
    - If the default **Group by Rule** is enabled, click the <i class="fa-regular fa-window-restore"></i> **Details** icon on the card of the finding.
```

        ![Click View details if Group by Rule is enabled](/img/cloud-platform-findings-group-by-rule-view-details.png#md-width)

>     - If **No grouping** view is enabled, click the **header hyperlink** on the card of the finding. In the example on the screenshot below, it is the **detected-generic-api-key**.

Reviewer: ```suggestion
    - If the **No grouping** view is enabled, click the **header hyperlink** on the card of the finding. In the example on the screenshot below, it is the **detected-generic-api-key**.
```

#### Add notes to findings

To **add notes** to the activity history of a finding:

1. Select a finding where you want to view details or add notes, and then do one of the following actions:
    - If the default **Group by Rule** is enabled, click <i class="fa-regular fa-window-restore"></i> **Details** icon on the card of the finding.
        ![Click View details if Group by Rule is enabled](/img/cloud-platform-findings-group-by-rule-view-details.png#md-width)
    - If **No grouping** view is enabled, click the **header hyperlink** on the card of the finding. In the example on the screenshot below, it is the **detected-generic-api-key**.
        ![Click View details if No grouping is enabled](/img/cloud-platform-findings-no-grouping-view-details.png#md-width)
2. View, or add the notes in the **Activity** section. To add a new note, click plus **New note**.
    ![Semgrep Cloud Platform finding details page](/img/cloud-platform-finding-details.png#md-width)

## Dataflow traces

<DisplayTaintedDataIntro />

### View dataflow traces

<DisplayTaintedDataProcedure />

## Data retention

> The retention period for findings are as follows:

Reviewer: ```suggestion
The retention period for findings is **5 years**.
```
Since this is always true we don't need a table. (The table previously included Community tier, which we sunset.)

| Retention period | Tier availability |

> | ---------------  | ----------------- |

Reviewer: ```suggestion
```

> | 5 years          | Team/Enterprise   |

Reviewer: ```suggestion
```
